Line/iOS/LINELoginDemo/LINELoginDemo/LoginController.m

```diff
@@ -0,0 +1,116 @@
+//
+//  ViewController.m
+//  LINELoginDemo
+//
+//  Created by Khanh LeViet on 10/5/16.
+//  Copyright (c) Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#import "LoginController.h"
+#import "LineAuthManager.h"
+#import <GTMHTTPFetcher.h>
+#import "HUDView.h"
+@import Firebase;
+
+@interface LoginController ()
+
+@property (weak, nonatomic) IBOutlet UIButton *loginButton;
+@property (weak, nonatomic) IBOutlet UIView *userView;
+@property (weak, nonatomic) IBOutlet UILabel *displayNameLabel;
+@property (weak, nonatomic) IBOutlet UIImageView *profilePictureView;
+@property (weak, nonatomic) HUDView *hudView;
+
+@property (nonatomic) FIRAuthStateDidChangeListenerHandle authStateChangeHandle;
+@property (nonatomic) GTMHTTPFetcher *imageFetcher;
+@property (nonatomic) BOOL isLoggingIn;
+
+@end
+
+@implementation LoginController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    
+    self.isLoggingIn = NO;
+    [self updateUIWithUser:[[FIRAuth auth] currentUser]];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
```

Comment:
Fixed in this [commit](https://github.com/firebase/custom-auth-samples/pull/1/commits/cfb5c9d622d615a16b4c27754be32543065571cc) :)